### PR TITLE
Improve calendar documentation

### PR DESCRIPTION
**Pull request by olliecurtis**
This PR adds additonal documentation on how the calendar component is constructed to help add clarity on the component

### CODE REVIEW

File: packages/bpk-component-calendar/DOCS.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+Default export is `BpkCalendarContainer` - This provides an out of the box calendar.
+
+HOCs - 
+  
+  - `withCalendarState` default export creates calendar using all our exports to construct component - which keeps track of the calendar state. In theory a custom state manager could be used for the calendar but would require a lot more effort to create and maintain
```
Comment (k0nserv) [inline packages/bpk-component-calendar/DOCS.md]: "create and maintain" To be clear this is for the consumer, we provide `withCalendarState` to make it easier for them

Comment (olliecurtis) [reply packages/bpk-component-calendar/DOCS.md]: Would `from a consumer perspective` be a good addition to add on at the end?

Comment (k0nserv) [reply packages/bpk-component-calendar/DOCS.md]: Yes 👍

Comment (olliecurtis) [reply packages/bpk-component-calendar/DOCS.md]: All done and added 😄

**Pull request merged**